template.php

```diff
@@ -372,12 +372,22 @@ function campaignion_foundation_form_alter(&$form, $form_state, $form_id) {
  * JavaScripts to pick up.
  */
 function campaignion_foundation_form_webform_client_form_alter(&$form, &$form_state, $form_id) {
-  $settings['campaignion_foundation']['webform'] = [
-    'total_steps' => $form_state['webform']['page_count'],
-    'current_step' => $form_state['webform']['page_num'],
-    'last_completed_step' => $form_state['webform']['page_visited'] ?? 0,
-  ];
-  $form['#attached']['js'][] = ['data' => $settings, 'type' => 'setting'];
+  // Check if the current node id is the node id the webform was attached to.
+  // This is needed to deal with the situation when there are multiple webforms
+  // on a single node.
+  // When the page nid and the webform's nid are the same we treat the form as
+  // the "main form" of the page. Only the main form determines the current
+  // step.
+  if ($node = menu_get_object()) {
+    if ($node->nid == $form['#node']->nid) {
```

Comment:
These two lines could be combined:

```php
if (($node = menu_get_object()) && $node->nid == $form['#node']->nid) {
}
```
… or if you want to do something more for readability:

```php
$is_current_nodes_form = ($node = menu_get_object()) && $node->nid == $form['#node']->nid;
ìf ($is_current_nodes_form) {
}
```